cobra/flux_analysis/parsimonious.py

```
@@ -11,53 +17,86 @@ def optimize_minimal_flux(model, already_irreversible=False,
+    cobra_model : :class:`~cobra.core.Model` object
 
     already_irreversible : bool, optional
         By default, the model is converted to an irreversible one.
         However, if the model is already irreversible, this step can be
-        skipped.
+        skipped
+
+    fraction_of_optimum : float, optional
+        Fraction of optimum which must be maintained. The original objective
+        reaction is constrained to be greater than maximal_value *
+        fraction_of_optimum. By default, this option is specified to be 1.0
+
+    desired_objective_value : float, optional
+        A desired objective value for the minimal solution that bypasses the
+        initial optimization result.
 
+    solver : string of solver name
+        If None is given, the default solver will be used.
+
+    Updates everything in-place, returns model to original state at end.
     """
-    if not already_irreversible:
-        modify.convert_to_irreversible(model)
-    model.optimize(**optimize_kwargs)
-    # if the problem is infeasible
-    if model.solution.f is None:
-        raise Exception("model could not be solved")
-    old_f = model.solution.f
-    old_objective_coefficients = {}
-    old_lower_bounds = {}
-    old_upper_bounds = {}
-    for reaction in model.reactions:
-        # if the reaction has a nonzero objective coefficient, then
-        # the same flux should be maintained through that reaction
+
+    if 'objective_sense' in optimize_kwargs:
+        if optimize_kwargs['objective_sense'] == 'minimize':
+            raise RuntimeWarning(
+                'Minimization not supported in optimize_minimal_flux')
+        optimize_kwargs.pop('objective_sense', None)
+
+    # Convert to irreversible, so all reactions will have a positive flux
+    convert_to_irreversible(cobra_model)
+
+    solver = solver_dict[get_solver_name() if solver is None else solver]
+
+    solution = None
+
+    # Maximize the existing objective function.

```

I would rewrite this block as follows
````
lp = solver.create_problem(cobra_model, **optimize_kwargs)
if not desired_objective_value:
    solver.solve_problem(lp, objective_sense='maximize')
    status = solver.get_status(lp)
    if status != "optimal":
        raise ValueError("pFBA requires optimal solution status, not " + solution.status)
    desired_objective_value = fraction_of_optimum * solver.get_objective_value(lp)
```

Then you set the objective to desired_objective_value / coefficient.

Also I think this function should raise an Exception when there is more than one objective, as discussed in this thread: https://groups.google.com/forum/#!topic/cobra-pie/AgbXPIznoIo